src/transport/mod.rs

```diff
@@ -78,7 +78,13 @@ pub enum FidoProtocol {
 }
 
 pub trait FidoDeviceIO {
-    fn send_msg<Out, Req: Request<Out>>(&mut self, msg: &Req) -> Result<Out, HIDError> {
+    fn send_msg<
+        Out,
+        Req: Request<Out> + RequestCtap1<Output = Out> + RequestCtap2<Output = Out>,
+    >(
```

Comment:
There's no reason to define `Request` if we're going to do this. We should fix the definition of `Request<T>` to make this more explicit, e.g:
```rust
pub trait Request<T>
 where
    Self: fmt::Debug,
    Self: RequestCtap1<Output=T>,
    Self: RequestCtap2<Output=T>,
 {
 }
```

Then it's clear that the new `AuthenticatorConfig` and `BioEnrollment` commands should not implement `Request`.

FWIW I mocked this up and got errors in `get_pin_uv_auth_param` and `determine_puap_if_needed`. Everything works if we change those to use `RequestCtap2<Output=V>` instead of `Request<V>`.